### ISSUE TITLE
2 FlipMoves on top of eachother

### ISSUE THREAD

**Issue by ConneXNL** [joshwcomeau/react-flip-move#105]
I am currently having a problem with 2 flip moves that are rendered on top of each other. The dom looks a bit like:
<ul>
   <li>
   <header>
      <FlipMove/>
    </header>
   </li>
   <li>
   <header>
      <FlipMove/>
    </header>
   </li>
</ul>

As soon as the first flip move grows in height (because an item gets added), the second flip move is unaffected, their parent just moves down (expected and wanted behavior). However, when the first flip move shrinks in height (because an item is removed), the items of the second flip move start animating up (although their parent already jumped up instantly).

I ran into the issue when I was using nested FlipMoves but apparently the problem already occurs without nesting flip moves.

Am I missing something?

**Comment by joshwcomeau** [comment]
Yeah, I wouldn't expect subsequent FlipMoves to animate when the earlier one changes.

Thanks for the examples. I'm pretty busy but I'll try and find the time to look into this in the next few days.

**Comment by joshwcomeau** [comment]
Hey @ConneXNL,

I finally found some time to look into it a bit. As far as I can tell, the animation _is_ consistent (although not ideal):

![stackedissue](https://cloud.githubusercontent.com/assets/6692932/21142409/f0bd7276-c10f-11e6-8cfd-d4b8676bfafd.gif)

It's possible that things have changed since the issue was open, we had a patch release in that time.

The nested example, however, is indeed pretty broken:
![nestedissue](https://cloud.githubusercontent.com/assets/6692932/21142504/801061d6-c110-11e6-925a-072e37140b18.gif)

If you or any other contributors want to take a shot at fixing it, I created a branch with the examples set up as stories: `fix/105-stacked-nested`. Otherwise, it may be a while before I find the time to fix it; hopefully the stacked version is workable for now!

(If you want "non-active" FlipMoves to not animate, I think I have an idea: I haven't tested it, but I suspect the problem might be solved if each FlipMove is in a separate component that uses `shouldComponentUpdate` to ensure that it isn't re-rendered when its props dont change. That way, the non-active FlipMove won't know that its position is changing)
